nbdime/webapp/nbdiffweb.py

```diff
@@ -45,13 +54,36 @@ def browse(port, base, remote, browser):
         DeprecationWarning)
 
 
+def is_gitref(candidate):
+    return is_valid_gitref(candidate) and (candidate is None or not os.path.exists(candidate))
```

Comment:
should probably reverse the priority so that if `os.path.exists(candidate)` is True, no git calls are made.